### PR TITLE
Extend keywords and snippets completion

### PR DESCRIPTION
**Pull request by henry-hsieh**
Currently, the system tasks ($) and compiler directives (`) are plaintext completion (InsertTextFormat). The completion for keywords is not provided. I extend system tasks and compiler directives to snippets completion and add full keywords as plaintext completion. I didn't add configurations to switch between plaintext and snippet completion for system tasks and compiler directives. It's may useful if LSP is slowed down by numerous completion candidates.

### CODE REVIEW

**Comment by kkanhere** [comment]
Thanks! I will take a look soon and get back to you.

File: src/svcompleter.ts [hunk 1]
```diff
@@ -29,133 +30,558 @@ import {
...
+    ["with"],
+    ["within"],
+    ["wor"],
+    ["xnor"],
+    ["xor"]
 ];
+let sv_completion_keyword_items = [];
+for (let n = 0; n < sv_completion_keywords.length; n++) {
+    sv_completion_keyword_items.push({
+        label: sv_completion_tick[n][0],
```
Comment (kkanhere) [inline src/svcompleter.ts:580]: Typo. Needs to be sv_completion_keywords

**Review by kkanhere — COMMENTED** [review]
Overall looks good. I can later verify if it impacts performance and accordingly introduce a setting.

File: src/svcompleter.ts (hunk 1 above)
Comment (henry-hsieh) [reply src/svcompleter.ts]: Fixed.

**Pull request merged**